src/widgets/MeterPanel.cpp

```diff
@@ -205,17 +204,19 @@ MeterUpdateQueue::~MeterUpdateQueue()
 
 void MeterUpdateQueue::Clear()
 {
-   mStart = 0;
-   mEnd = 0;
+   mStart.store(0);
+   mEnd.store(0);
 }
 
 // Add a message to the end of the queue.  Return false if the
 // queue was full.
 bool MeterUpdateQueue::Put(MeterUpdateMsg &msg)
 {
+   auto start = mStart.load(std::memory_order_acquire);
+   auto end = mEnd.load(std::memory_order_relaxed);
    // mStart can be greater than mEnd because it is all mod mBufferSize
-   wxASSERT( (mEnd + mBufferSize - mStart) >= 0 );
-   int len = (mEnd + mBufferSize - mStart) % mBufferSize;
+   assert( (end + mBufferSize - start) >= 0 );
```

Comment:
One, yes, it's good to reduce dependencies on wxWidgets where we can, though less important for some parts of widgets.

Two, in case the assertion failes, wxASSERT puts up a dialog box with a stack trace.  But dialogs should not be made from a worker thread.

Really I think assertions are supposed to be provable conditions that assist documentation of the program.  The proof of them should not require examining a large amount of code (that is why encapsulation is important) but may assume preconditions of functions have been honored by callers (and that is why it is important to state preconditions).